core/os/file_windows.odin

```diff
@@ -223,11 +225,13 @@ file_size :: proc(fd: Handle) -> (i64, Error) {
 MAX_RW :: 1<<30
 
 @(private)
-pread :: proc(fd: Handle, data: []byte, offset: i64) -> (int, Error) {
+pread :: proc(fd: Handle, data: []byte, offset: i64) -> (n: int, err: Error) {
+	curr_off := seek(fd, 0, 1) or_return
+	defer seek(fd, curr_off, 0)
```

Comment:
One thing this does do is make this not thread safe

Reply:
We would need a more elaborate `Handle` with a mutex then, wouldn't we?

Reply:
I'd worry less about this at the moment and make sure it is fine in `os2.File`

Reply:
So there are 2 bad solutions we can pick: either have consistent behavior across targets but not thread safe, or inconsistent behavior. I am leaning to your no thread safe solution with a comment to clarify that this isn't thread safe.

Reply:
`os2` file descriptors are now being tested. I have to resolve some issues with Darwin and Windows yet, but it's getting there.